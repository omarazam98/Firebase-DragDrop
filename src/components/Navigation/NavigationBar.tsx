import React from 'react';
import { withRouter, Link } from 'react-router-dom';
import { MenuList, MenuItem } from '@material-ui/core';
import { makeStyles, createStyles, Theme } from '@material-ui/core/styles';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    root: {
      display: 'flex',
    },
    paper: {
      marginRight: theme.spacing(2),
    },
    content: {
      paddingLeft: theme.spacing(6),
    },
  }),
);
interface NavigationBarProps {
  location: any;
  routesList: any;
}

export const NavigationBar: React.FC<NavigationBarProps> = (props) => {
  useStyles();
  return (
    <MenuList>
      {props.routesList.map((route: any, key: number) => {
        return (
          <MenuItem component={Link} to={route.path}
                    selected={props.location.pathname === route.path}
                    key={key}>
            {route.name}<br/>
          </MenuItem>
        );
      })}
    </MenuList>
  );
};

export default withRouter(NavigationBar);
