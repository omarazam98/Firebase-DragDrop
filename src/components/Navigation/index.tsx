// Main routing component.
// If a new page is added, be sure to add the route to src/constants/routes
import React from 'react';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';
import Routes from '../../constants/routes';
import {withAPI} from '@winwin/api-firebase';
import SplitPane from 'react-split-pane';

export function Navigation(props) {
  return (
    <div>
      <Router>
        <SplitPane split="vertical" minSize={50} defaultSize={100}>
          {/* Link creates the object that a user can click on to go to another page */}
          <div>
            {links()}
            <button onClick = {() => {props.api.auth.signOut()}}>Log Out</button>
          </div>
          {/* Route indicates what component should be shown, based on what is linked */}
          {/* Map routes from links to their components */}
          <div>
          {Routes.map((route:any, index) => (
            <Route
              key={index}
              path={route.path}
              exact={route.exact}
              component={route.component}
            />
          ))}
          </div>
        </SplitPane>
      </Router>
    </div>
  );
}
// iterate over all routes from routes.ts
// return a jsx expression containing links to all routes
function links() {
  // can't push JSX on an empty array, so init with a div
  const allLinks = [<div />];
  Routes.forEach((route:any) => allLinks.push(<Link to={route.path}>{route.name}<br /></Link>));
  return allLinks;
}

export default withAPI(Navigation);
