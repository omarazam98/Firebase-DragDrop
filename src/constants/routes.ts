// places routes in this file
// each page should have a route (e.g. winwinhomesharing.org/signup would have the route 'signup'

export const LANDING = '/';
