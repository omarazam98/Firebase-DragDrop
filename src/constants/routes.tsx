// places routes in this file
// each page/component should have a route
// e.g. winwinhomesharing.org/signup would have the route 'signup'
import SignUp from '../components/SignUp/SignUp'
import Login from '../components/Login/Login';
import Dashboard from '../components/Dashboard/Dashboard';
import EmailRedirect from '../components/EmailRedirect';

// to add a new page, simply add another element to the array below
// path is the path to the page
// exact is if the route is to the exact path
// name will be the label of the link
// component should be the page/component to be shown
// To add a componenet page to the routes table/nav bar, you should import
// it here at the top of the file and add it to the routes object:
/*** Example:
 * import App from '../../App'
 *
 * const ROUTES = [
 *  {
 *     path:'/',
 *     exact:true,
 *     name:'Home',
 *     component: App,
 *  },
/* Example routes with basic JSX elements as components
 *  {
 *     path:'/example1',
 *     name:'Primary',
 *     component: () => <div>Hello world!</div>,
 *  },
 *  {
 *     path:'/example2',
 *     exact:true,
 *     name:'Secondary',
 *     component: () => <div>Goodbye, world!</div>,
 *  },
 * ];
 *
 **/

/*These show up in the nav bar as links*/
export const NAVBAR_ROUTES = [
  {
    path: '/signup',
    exact: true,
    name: 'Sign Up',
    component: SignUp,
  },
  {
    path: '/login',
    exact: true,
    name: 'Log In',
    component: Login,
  },
  {
    path: '/dashboard',
    exact: true,
    name: 'Dashboard',
    component: Dashboard,
    authRequired: true,
  },
];

/*These are not rendered as links*/
export const ROUTES = [
  ...NAVBAR_ROUTES,
  {
    path: '/email',
    exact: true,
    name: 'Email Link',
    component: EmailRedirect,
  },
];
