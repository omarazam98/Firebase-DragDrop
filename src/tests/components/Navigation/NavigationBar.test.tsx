import React from 'react';
import enzyme, { shallow, mount } from 'enzyme';
import { NavigationBar } from '../../../components/Navigation/NavigationBar';
import { NAVBAR_ROUTES } from '../../../constants/routes';
import { MenuItem } from '@material-ui/core';
import { BrowserRouter as Router, Route, Link } from 'react-router-dom';

const mockLocation = {
  pathname: './dashboard',
};

test('Navbar Renders correctly', () => {
  const wrapper = mount(
    <Router>
      <NavigationBar location={mockLocation} routesList={NAVBAR_ROUTES} />
    </Router>,
  );
  expect(wrapper).toMatchSnapshot();
  const expectedLength = NAVBAR_ROUTES.length;
  const menuItems = wrapper.find(MenuItem).length;
  expect(menuItems).toEqual(expectedLength);
});
