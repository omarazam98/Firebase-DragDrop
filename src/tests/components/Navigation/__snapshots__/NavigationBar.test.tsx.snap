// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Navbar Renders correctly 1`] = `
<BrowserRouter>
  <Router
    history={
      Object {
        "action": "POP",
        "block": [Function],
        "createHref": [Function],
        "go": [Function],
        "goBack": [Function],
        "goForward": [Function],
        "length": 1,
        "listen": [Function],
        "location": Object {
          "hash": "",
          "pathname": "/",
          "search": "",
          "state": undefined,
        },
        "push": [Function],
        "replace": [Function],
      }
    }
  >
    <NavigationBar
      location={
        Object {
          "pathname": "./dashboard",
        }
      }
      routesList={
        Array [
          Object {
            "authRequired": false,
            "component": [Function],
            "exact": true,
            "name": "Sign Up",
            "path": "/signup",
          },
          Object {
            "authRequired": false,
            "component": [Function],
            "exact": true,
            "name": "Log In",
            "path": "/login",
          },
          Object {
            "authRequired": true,
            "component": [Function],
            "exact": true,
            "name": "Dashboard",
            "path": "/dashboard",
          },
        ]
      }
    >
      <ForwardRef(MenuList)>
        <WithStyles(ForwardRef(List))
          onKeyDown={[Function]}
          role="menu"
          tabIndex={-1}
        >
          <ForwardRef(List)
            classes={
              Object {
                "dense": "MuiList-dense",
                "padding": "MuiList-padding",
                "root": "MuiList-root",
                "subheader": "MuiList-subheader",
              }
            }
            onKeyDown={[Function]}
            role="menu"
            tabIndex={-1}
          >
            <ul
              className="MuiList-root MuiList-padding"
              onKeyDown={[Function]}
              role="menu"
              tabIndex={-1}
            >
              <WithStyles(ForwardRef(MenuItem))
                component={[Function]}
                key="0"
                selected={false}
                to="/signup"
              >
                <ForwardRef(MenuItem)
                  classes={
                    Object {
                      "gutters": "MuiMenuItem-gutters",
                      "root": "MuiMenuItem-root",
                      "selected": "Mui-selected",
                    }
                  }
                  component={[Function]}
                  selected={false}
                  to="/signup"
                >
                  <WithStyles(ForwardRef(ListItem))
                    button={true}
                    className="MuiMenuItem-root MuiMenuItem-gutters"
                    component={[Function]}
                    disableGutters={false}
                    role="menuitem"
                    selected={false}
                    tabIndex={-1}
                    to="/signup"
                  >
                    <ForwardRef(ListItem)
                      button={true}
                      className="MuiMenuItem-root MuiMenuItem-gutters"
                      classes={
                        Object {
                          "alignItemsFlexStart": "MuiListItem-alignItemsFlexStart",
                          "button": "MuiListItem-button",
                          "container": "MuiListItem-container",
                          "dense": "MuiListItem-dense",
                          "disabled": "Mui-disabled",
                          "divider": "MuiListItem-divider",
                          "focusVisible": "Mui-focusVisible",
                          "gutters": "MuiListItem-gutters",
                          "root": "MuiListItem-root",
                          "secondaryAction": "MuiListItem-secondaryAction",
                          "selected": "Mui-selected",
                        }
                      }
                      component={[Function]}
                      disableGutters={false}
                      role="menuitem"
                      selected={false}
                      tabIndex={-1}
                      to="/signup"
                    >
                      <WithStyles(ForwardRef(ButtonBase))
                        className="MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                        component={[Function]}
                        disabled={false}
                        focusVisibleClassName="Mui-focusVisible"
                        role="menuitem"
                        tabIndex={-1}
                        to="/signup"
                      >
                        <ForwardRef(ButtonBase)
                          className="MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                          classes={
                            Object {
                              "disabled": "Mui-disabled",
                              "focusVisible": "Mui-focusVisible",
                              "root": "MuiButtonBase-root",
                            }
                          }
                          component={[Function]}
                          disabled={false}
                          focusVisibleClassName="Mui-focusVisible"
                          role="menuitem"
                          tabIndex={-1}
                          to="/signup"
                        >
                          <Link
                            aria-disabled={false}
                            className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                            onBlur={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            role="menuitem"
                            tabIndex={-1}
                            to="/signup"
                          >
                            <a
                              aria-disabled={false}
                              className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                              href="/signup"
                              onBlur={[Function]}
                              onClick={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              role="menuitem"
                              tabIndex={-1}
                            >
                              Sign Up
                              <br
                                key=".1"
                              />
                              <NoSsr>
                                <WithStyles(TouchRipple)
                                  center={false}
                                >
                                  <TouchRipple
                                    center={false}
                                    classes={
                                      Object {
                                        "child": "MuiTouchRipple-child",
                                        "childLeaving": "MuiTouchRipple-childLeaving",
                                        "childPulsate": "MuiTouchRipple-childPulsate",
                                        "ripple": "MuiTouchRipple-ripple",
                                        "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                        "rippleVisible": "MuiTouchRipple-rippleVisible",
                                        "root": "MuiTouchRipple-root",
                                      }
                                    }
                                  >
                                    <span
                                      className="MuiTouchRipple-root"
                                    >
                                      <TransitionGroup
                                        childFactory={[Function]}
                                        component={null}
                                        enter={true}
                                        exit={true}
                                      />
                                    </span>
                                  </TouchRipple>
                                </WithStyles(TouchRipple)>
                              </NoSsr>
                            </a>
                          </Link>
                        </ForwardRef(ButtonBase)>
                      </WithStyles(ForwardRef(ButtonBase))>
                    </ForwardRef(ListItem)>
                  </WithStyles(ForwardRef(ListItem))>
                </ForwardRef(MenuItem)>
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                component={[Function]}
                key="1"
                selected={false}
                to="/login"
              >
                <ForwardRef(MenuItem)
                  classes={
                    Object {
                      "gutters": "MuiMenuItem-gutters",
                      "root": "MuiMenuItem-root",
                      "selected": "Mui-selected",
                    }
                  }
                  component={[Function]}
                  selected={false}
                  to="/login"
                >
                  <WithStyles(ForwardRef(ListItem))
                    button={true}
                    className="MuiMenuItem-root MuiMenuItem-gutters"
                    component={[Function]}
                    disableGutters={false}
                    role="menuitem"
                    selected={false}
                    tabIndex={-1}
                    to="/login"
                  >
                    <ForwardRef(ListItem)
                      button={true}
                      className="MuiMenuItem-root MuiMenuItem-gutters"
                      classes={
                        Object {
                          "alignItemsFlexStart": "MuiListItem-alignItemsFlexStart",
                          "button": "MuiListItem-button",
                          "container": "MuiListItem-container",
                          "dense": "MuiListItem-dense",
                          "disabled": "Mui-disabled",
                          "divider": "MuiListItem-divider",
                          "focusVisible": "Mui-focusVisible",
                          "gutters": "MuiListItem-gutters",
                          "root": "MuiListItem-root",
                          "secondaryAction": "MuiListItem-secondaryAction",
                          "selected": "Mui-selected",
                        }
                      }
                      component={[Function]}
                      disableGutters={false}
                      role="menuitem"
                      selected={false}
                      tabIndex={-1}
                      to="/login"
                    >
                      <WithStyles(ForwardRef(ButtonBase))
                        className="MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                        component={[Function]}
                        disabled={false}
                        focusVisibleClassName="Mui-focusVisible"
                        role="menuitem"
                        tabIndex={-1}
                        to="/login"
                      >
                        <ForwardRef(ButtonBase)
                          className="MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                          classes={
                            Object {
                              "disabled": "Mui-disabled",
                              "focusVisible": "Mui-focusVisible",
                              "root": "MuiButtonBase-root",
                            }
                          }
                          component={[Function]}
                          disabled={false}
                          focusVisibleClassName="Mui-focusVisible"
                          role="menuitem"
                          tabIndex={-1}
                          to="/login"
                        >
                          <Link
                            aria-disabled={false}
                            className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                            onBlur={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            role="menuitem"
                            tabIndex={-1}
                            to="/login"
                          >
                            <a
                              aria-disabled={false}
                              className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                              href="/login"
                              onBlur={[Function]}
                              onClick={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              role="menuitem"
                              tabIndex={-1}
                            >
                              Log In
                              <br
                                key=".1"
                              />
                              <NoSsr>
                                <WithStyles(TouchRipple)
                                  center={false}
                                >
                                  <TouchRipple
                                    center={false}
                                    classes={
                                      Object {
                                        "child": "MuiTouchRipple-child",
                                        "childLeaving": "MuiTouchRipple-childLeaving",
                                        "childPulsate": "MuiTouchRipple-childPulsate",
                                        "ripple": "MuiTouchRipple-ripple",
                                        "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                        "rippleVisible": "MuiTouchRipple-rippleVisible",
                                        "root": "MuiTouchRipple-root",
                                      }
                                    }
                                  >
                                    <span
                                      className="MuiTouchRipple-root"
                                    >
                                      <TransitionGroup
                                        childFactory={[Function]}
                                        component={null}
                                        enter={true}
                                        exit={true}
                                      />
                                    </span>
                                  </TouchRipple>
                                </WithStyles(TouchRipple)>
                              </NoSsr>
                            </a>
                          </Link>
                        </ForwardRef(ButtonBase)>
                      </WithStyles(ForwardRef(ButtonBase))>
                    </ForwardRef(ListItem)>
                  </WithStyles(ForwardRef(ListItem))>
                </ForwardRef(MenuItem)>
              </WithStyles(ForwardRef(MenuItem))>
              <WithStyles(ForwardRef(MenuItem))
                component={[Function]}
                key="2"
                selected={false}
                to="/dashboard"
              >
                <ForwardRef(MenuItem)
                  classes={
                    Object {
                      "gutters": "MuiMenuItem-gutters",
                      "root": "MuiMenuItem-root",
                      "selected": "Mui-selected",
                    }
                  }
                  component={[Function]}
                  selected={false}
                  to="/dashboard"
                >
                  <WithStyles(ForwardRef(ListItem))
                    button={true}
                    className="MuiMenuItem-root MuiMenuItem-gutters"
                    component={[Function]}
                    disableGutters={false}
                    role="menuitem"
                    selected={false}
                    tabIndex={-1}
                    to="/dashboard"
                  >
                    <ForwardRef(ListItem)
                      button={true}
                      className="MuiMenuItem-root MuiMenuItem-gutters"
                      classes={
                        Object {
                          "alignItemsFlexStart": "MuiListItem-alignItemsFlexStart",
                          "button": "MuiListItem-button",
                          "container": "MuiListItem-container",
                          "dense": "MuiListItem-dense",
                          "disabled": "Mui-disabled",
                          "divider": "MuiListItem-divider",
                          "focusVisible": "Mui-focusVisible",
                          "gutters": "MuiListItem-gutters",
                          "root": "MuiListItem-root",
                          "secondaryAction": "MuiListItem-secondaryAction",
                          "selected": "Mui-selected",
                        }
                      }
                      component={[Function]}
                      disableGutters={false}
                      role="menuitem"
                      selected={false}
                      tabIndex={-1}
                      to="/dashboard"
                    >
                      <WithStyles(ForwardRef(ButtonBase))
                        className="MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                        component={[Function]}
                        disabled={false}
                        focusVisibleClassName="Mui-focusVisible"
                        role="menuitem"
                        tabIndex={-1}
                        to="/dashboard"
                      >
                        <ForwardRef(ButtonBase)
                          className="MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                          classes={
                            Object {
                              "disabled": "Mui-disabled",
                              "focusVisible": "Mui-focusVisible",
                              "root": "MuiButtonBase-root",
                            }
                          }
                          component={[Function]}
                          disabled={false}
                          focusVisibleClassName="Mui-focusVisible"
                          role="menuitem"
                          tabIndex={-1}
                          to="/dashboard"
                        >
                          <Link
                            aria-disabled={false}
                            className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                            onBlur={[Function]}
                            onDragLeave={[Function]}
                            onFocus={[Function]}
                            onKeyDown={[Function]}
                            onKeyUp={[Function]}
                            onMouseDown={[Function]}
                            onMouseLeave={[Function]}
                            onMouseUp={[Function]}
                            onTouchEnd={[Function]}
                            onTouchMove={[Function]}
                            onTouchStart={[Function]}
                            role="menuitem"
                            tabIndex={-1}
                            to="/dashboard"
                          >
                            <a
                              aria-disabled={false}
                              className="MuiButtonBase-root MuiListItem-root MuiMenuItem-root MuiMenuItem-gutters MuiListItem-gutters MuiListItem-button"
                              href="/dashboard"
                              onBlur={[Function]}
                              onClick={[Function]}
                              onDragLeave={[Function]}
                              onFocus={[Function]}
                              onKeyDown={[Function]}
                              onKeyUp={[Function]}
                              onMouseDown={[Function]}
                              onMouseLeave={[Function]}
                              onMouseUp={[Function]}
                              onTouchEnd={[Function]}
                              onTouchMove={[Function]}
                              onTouchStart={[Function]}
                              role="menuitem"
                              tabIndex={-1}
                            >
                              Dashboard
                              <br
                                key=".1"
                              />
                              <NoSsr>
                                <WithStyles(TouchRipple)
                                  center={false}
                                >
                                  <TouchRipple
                                    center={false}
                                    classes={
                                      Object {
                                        "child": "MuiTouchRipple-child",
                                        "childLeaving": "MuiTouchRipple-childLeaving",
                                        "childPulsate": "MuiTouchRipple-childPulsate",
                                        "ripple": "MuiTouchRipple-ripple",
                                        "ripplePulsate": "MuiTouchRipple-ripplePulsate",
                                        "rippleVisible": "MuiTouchRipple-rippleVisible",
                                        "root": "MuiTouchRipple-root",
                                      }
                                    }
                                  >
                                    <span
                                      className="MuiTouchRipple-root"
                                    >
                                      <TransitionGroup
                                        childFactory={[Function]}
                                        component={null}
                                        enter={true}
                                        exit={true}
                                      />
                                    </span>
                                  </TouchRipple>
                                </WithStyles(TouchRipple)>
                              </NoSsr>
                            </a>
                          </Link>
                        </ForwardRef(ButtonBase)>
                      </WithStyles(ForwardRef(ButtonBase))>
                    </ForwardRef(ListItem)>
                  </WithStyles(ForwardRef(ListItem))>
                </ForwardRef(MenuItem)>
              </WithStyles(ForwardRef(MenuItem))>
            </ul>
          </ForwardRef(List)>
        </WithStyles(ForwardRef(List))>
      </ForwardRef(MenuList)>
    </NavigationBar>
  </Router>
</BrowserRouter>
`;
